scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -1085,33 +1087,55 @@ class FormatWriter(formatOps: FormatOps) {
     }
 
     lazy val extraBlankTokens = {
-      val extraBlankBuffer = Set.newBuilder[Int]
-      def setFt(ft: FormatToken): Unit =
-        extraBlankBuffer += ft.meta.idx
+      val extraBlankMap = new mutable.HashMap[Int, Int]
+      def setFt(ft: FormatToken, cnt: Int = 1): Unit = if (cnt > 0) {
+        val idx = ft.meta.idx
+        if (extraBlankMap.getOrElseUpdate(idx, cnt) < cnt)
+          extraBlankMap.update(idx, cnt)
+      }
       def indentedPackage(pkg: Pkg) = tokens.tokenAfter(pkg.ref).right match {
         case _: T.LeftBrace | _: T.Colon => true
         case _ => false
       }
-      def setTopStats(owner: Tree, stats: Seq[Tree]): Unit = {
+      @tailrec
+      def getNest(tree: Tree, curNest: Int): Int = tree.parent match {
+        case Some(_: Source) | None => curNest
+        case Some(t: Pkg) if !indentedPackage(t) => getNest(t, curNest)
+        case Some(t) => getNest(t, curNest + 1)
+      }
+      def setTopStats(owner: Tree, stats: Seq[Stat], curNest: Int = 1): Unit = {
+        if (stats.isEmpty) return
+        val nest = getNest(owner, curNest)
+        if (nest < 0) return
```

Comment:
When will nest be smaller than 0 ? It seems it's only incremented.

Reply:
I used to have a check for `Term.Block` there, which returned -1, but removed it as unnecessary (see comment below). however, in the future, it's possible that we'll add another check which will prohibit nesting by returning a negative value.